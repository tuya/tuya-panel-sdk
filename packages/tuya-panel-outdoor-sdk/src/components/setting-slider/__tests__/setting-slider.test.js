import React from 'react';
import { shallow } from 'enzyme';
import SettingSlider from '../index';

describe('SettingSlider components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <SettingSlider
        code="control"
        range={['open', 'stop', 'close']}
        value="stop"
        step={1}
        themeColor="#57BCFB"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
