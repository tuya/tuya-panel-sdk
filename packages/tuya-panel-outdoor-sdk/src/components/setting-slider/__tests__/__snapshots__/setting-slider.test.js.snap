// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingSlider components basic render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Object {
        "color": "#999",
        "fontSize": 32,
        "marginHorizontal": 24,
        "paddingTop": 88,
      }
    }
    type={null}
    weight={null}
  />
  <View
    style={
      Object {
        "paddingTop": 104,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 88,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "left": 60,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 24,
              "display": "flex",
              "height": 72,
              "justifyContent": "center",
              "width": 64,
            },
            Object {
              "backgroundColor": "#57BCFB",
            },
          ]
        }
      >
        <TYText
          align={null}
          color="#fff"
          numberOfLines={2}
          size={null}
          type={null}
          weight={null}
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderWidth": 8,
              "bottom": 0,
              "left": 68,
              "position": "absolute",
            },
            Object {
              "borderTopColor": "#57BCFB",
            },
          ]
        }
      />
    </View>
    <Component
      canTouchTrack={true}
      maximumTrackTintColor="#eee"
      maximumValue={2}
      minimumTrackTintColor="#57BCFB"
      minimumValue={0}
      onSlidingComplete={[Function]}
      onValueChange={[Function]}
      stepValue={1}
      style={
        Object {
          "backgroundColor": "#eee",
          "borderRadius": 10,
          "height": 20,
          "width": 207,
        }
      }
      trackStyle={
        Object {
          "borderRadius": 6,
          "height": 12,
          "margin": 4,
        }
      }
      value={1}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Object {
        "color": "#999",
        "fontSize": 32,
        "marginHorizontal": 24,
        "paddingTop": 88,
      }
    }
    type={null}
    weight={null}
  />
</View>
`;
